### PR TITLE
Placeable lava and water when using buckets and get an empty bucket in y...

### PR DESCRIPTION
**Pull request by Darsstar**
...our inventory.

Scoop up lava and water using buckets and get a water/lava bucket in your inventory.
Added Override annotations.
Refactored the gamemode command.

I don't really like the names of the new classes...
It's more a proof of concept kind of thing to me, so I or somebody else can improve on it later.

Let me know what needs to be changed to merge this.

PS. are this to many things at once for one PR?

### CODE REVIEW

**Comment by Zidane** [comment]
Need for you to rebase this with the HEAD before I can pull it in :P

Also it isn't too many things as it is mostly small things and a couple of big changes. If unsure in the future, multiple PRs are no issue to handle so feel free.

**Comment by Darsstar** [comment]
Done, I think...
Bit sick and light in the head, so I wouldn't be surprised if I did something wrong...

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java [hunk 1]
```diff
@@ -45,71 +49,66 @@
...
+			if (!(source instanceof Player)) {
+				throw new CommandException("You must be a player to toggle your game mode.");
+			}
+
+			player = (Player)source;
+		}
+
+		int mode;
+
+		if (args.isInteger(index)) {
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java]: Oh good, this got added ^_^

**Comment by Zidane** [comment]
@Darsstar just waiting for you to rebase this so I can auto pull it ;).

**Comment by Darsstar** [comment]
I did not want to close this github!

**Pull request merged**